validator/src/bootstrap.rs

```diff
@@ -1115,19 +1117,32 @@ fn download_snapshots(
         )?;
     }
 
-    // Check and see if we've already got the incremental snapshot; if not, download it
-    if let Some(incremental_snapshot_hash) = incremental_snapshot_hash {
+    // If the full snapshot download took more than a few minutes there is probably a newer incremental snapshot
+    let newest_known_incremental_snapshot = match check_for_newer_incremental_snapshot(
```

Comment:
I would use shadowing here, we don't need original variable below.
```
let incremental_snapshot_hash = match check_for_newer_incremental_snapshot(
...
```
And it'll reduce PR too.

PS. I don't have write permission for suggesting.